target_chains/ethereum/contracts/contracts/pulse/IPulse.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.0;
+
+import "./PulseEvents.sol";
+import "./PulseState.sol";
+
+interface IPulseConsumer {
+    function pulseCallback(
```

Comment:
we had another function here to get the provider. Otherwise it was pretty easy for users to shoot themselves in the foot and accept this callback from everywhere.

Reply:
Please take a look at this file to see how it's handled there:
https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/ethereum/entropy_sdk/solidity/IEntropyConsumer.sol

Reply:
will remove provider altogether since its not important for Pulse